8.x/core-concepts/annotations.md

```diff
@@ -2,4 +2,423 @@
+
+**OUTPUT**
+
+![mere-comment](/img-annotations/04.png)
+
+The output returns a mere comment. In order to get a processed result, change the code in the `public function handle`:
+
+```php{4}
+public function handle()
+{
+    $testAnnotationsRef = new ReflectiveClass(TestAnnotations::class);
+    dd($testAnnotationsRef->getAnnotations());
+}
+```
+
+**OUTPUT**
+
+![processed-comment](/img-annotations/05.png)
+
+As you see, the comment is processed and represented in detail.
+
+Now change the DocBloc above the `TestAnnotations class`:
+
+```php{2}
+/**
+ * @salam(Ali="2", Test=3)
+ */
+class TestAnnotations extends Command
+{
+    ...
+}
+```
+
+**OUTPUT**
+
+![accurate-result](/img-annotations/06.png)
+
+As you see, the result is accurate. So you can write some descriptions above a class and then use them in your code.
+
+Now make a new method called `testFunction` with its related DocBlock into the `TestAnnotations class`:
+
+```php{9-15}
+class TestAnnotations extends Command
+{
+    ...
+        public function __construct()
+    {
+        parent::__construct();
+    }
+    
+    /**
+     * @rules(username="required", password="required|max:20")
+     */
+    public function testFunction()
+    {
+        echo "salam";
+    }
+    ...
+}
+```
+
+Put the codes below in the path `/larammerce/app/Console/Commands/TestAnnotations.php`:
+
+```php{6,14,15}
+<?php
+
+namespace App\Console\Commands;
+use Illuminate\Console\Command;
+use App\Utils\Reflection\ReflectiveClass;
+use App\Utils\Reflection\ReflectiveMethod;
+
+class TestAnnotations extends Command
+{
+    ...
+    public function handle()
+    {
+        $testAnnotationsRef = new ReflectiveClass(TestAnnotations::class);
+        $testFunctionRef = new ReflectiveMethod(TestAnnotations::class, "testFunction");
+        dd($testFunctionRef->hasAnnotation("salam"));
+    }
+}
+```
+
+**OUTPUT**
+
+![no-exist-salam](/img-annotations/07.png)
+
+This means that there is no annotation named `salam` above the `testFunction` method.
+
+Now change this line into the `public function handle`:
+
+```php{4}
+public function handle()
+{
+    ...
+    dd($testFunctionRef->hasAnnotation("rules"));
+}
+```
+
+**OUTPUT**
+
+![exist-rules](/img-annotations/08.png)
+
+It declares that an annotation named `rules` exists above the `testFunction` method.
+
+To get this annotation, insert these codes into the `public function handle`:
+
+```php{5-7}
+public function handle()
+{
+    $testAnnotationsRef = new ReflectiveClass(TestAnnotations::class);
+    $testFunctionRef = new ReflectiveMethod(TestAnnotations::class, "testFunction");
+    if($testFunctionRef->hasAnnotation("rules")){
+        dd($testFunctionRef->getAnnotation("rules"));
+    }
+}
+```
+
+**OUTPUT**
+
+![test-function](/img-annotations/09.png)
+
+As you see, the output shows the DocBlock above the `testFunction` method.
+
+You can also do mathematical calculations into the DocBlock.For example, change the DocBlock above the `testFunction` method:
```

Comment:
Docblock. For --> put a space at the first of the line.

Reply:
Done.